### PR TITLE
Expose context property on State object.

### PR DESCRIPTION
**Pull request by asherw**

### CODE REVIEW

File: src/HassClient/Client/Messages.cs [hunk 1]
```diff
@@ -100,6 +100,14 @@ public class HassState
         [JsonPropertyName("last_changed")] public DateTime LastChanged { get; set; } = DateTime.MinValue;
         [JsonPropertyName("last_updated")] public DateTime LastUpdated { get; set; } = DateTime.MinValue;
         [JsonPropertyName("state")] public dynamic? State { get; set; } = "";
+        [JsonPropertyName("context")] public HassContext? Context { get; set; } = null;
+    }
+
+    public class HassContext
+    {
+        [JsonPropertyName("id")] public string Id { get; set; }
```
Comment (helto4real) [inline src/HassClient/Client/Messages.cs:108]: Set default values please to remove warnings. We do warning free code as a code quality measure. Should be easy.  integer to 0, strings to empty strings and objects to nullable types that sets to null. 
Thank you for the contribution. Other than this looks good.

File: src/HassClient/Client/Messages.cs [hunk 2]
```diff
@@ -275,7 +283,6 @@ public static dynamic ToDynamic(this JsonElement element)
             {
                 foreach (var obj in element.EnumerateObject())
                 {
-                    var jsonElem = obj.Value;
```
Comment (helto4real) [inline src/HassClient/Client/Messages.cs:278]: Nice find... must have forgot them :)

File: src/HassClient/Client/Messages.cs [hunk 3]
```diff
@@ -359,7 +366,6 @@ public static IEnumerable<HassServiceDomain> ToServicesResult(this JsonElement e
 
             foreach (var property in element.EnumerateObject())
             {
-                var servicesList = new List<HassService>();
```
Comment (helto4real) [inline src/HassClient/Client/Messages.cs:362]: Nice find... must have forgot them :)

File: tests/HassClient.Unit.Tests/HassClientTests.cs [hunk 4]
```diff
@@ -791,6 +789,9 @@ public async Task SubscribeToEventsReturnsCorrectEvent()
 
             // Just test one of the NewStateOne
             Assert.True(stateMessage.NewState?.EntityId == "binary_sensor.vardagsrum_pir");
+
+            Assert.NotNull(stateMessage.OldState.Context);
+            Assert.NotNull(stateMessage.NewState.Context);
```
Comment (helto4real) [inline tests/HassClient.Unit.Tests/HassClientTests.cs:794]: Thanks for adding test also!

**Comment by asherw** [comment]
Thanks for the feedback.  Changes made.

Cheers!

**Pull request merged**